sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -91,7 +91,7 @@ case class EqualTo(attribute: String, value: Any) extends Filter {
   override def toV2: Predicate = {
     val literal = Literal(value)
     new Predicate("=",
-      Array(FieldReference(attribute), LiteralValue(literal.value, literal.dataType)))
+      Array(FieldReference.column(attribute), LiteralValue(literal.value, literal.dataType)))
```

Comment:
The column name in v1 filter is generated by `PushableColumn`, and we need to match it.

When nested predicate pushdown is enabled, we already quote the column name if the name is not a valid SQL identifier. So `FieldReference.apply` is correct here. However, when it's disabled, the column name is the parsed name and `FieldReference.column` should be used.

I think what we can do is
```
try {
  FieldReference(attribute)
} catch {
  case _: ParseException => FieldReference.column(attribute)
}
```

Reply:
We can add a `protected def toV2Column` in the base class of v1 filter

Reply:
Thanks for the suggestion! Fixed.